contracts/libraries/Ticks.sol

```diff
@@ -39,17 +39,20 @@ library Ticks {
     function initialize(
         IRangePoolStructs.TickMap storage tickMap,
         IRangePoolStructs.Sample[65535] storage samples,
-        IRangePoolStructs.PoolState memory state
+        IRangePoolStructs.PoolState memory state,
+        int24 tickSpacing
     ) external returns (
         IRangePoolStructs.PoolState memory
     )    
     {
-        TickMap.set(tickMap, TickMath.MIN_TICK);
-        TickMap.set(tickMap, TickMath.MAX_TICK);
+        uint160 minPrice = TickMath.getSqrtRatioAtTick(TickMap._round(TickMath.MIN_TICK, tickSpacing));
+        uint160 maxPrice = TickMath.getSqrtRatioAtTick(TickMap._round(TickMath.MAX_TICK, tickSpacing));
+        if (state.price < minPrice || state.price >= maxPrice) require(false, 'StartPriceInvalid()');
```

Comment:
`startPrice` validation has been moved here since the actual MIN_TICK and MAX_TICK is based on `tickSpacing`.